src/views/Account/Rewards/RewardsCard.tsx

```diff
@@ -0,0 +1,834 @@
+import { PrizeWLaurels } from '@components/Images/PrizeWithLaurels'
+import { LoadingList } from '@components/PrizePoolDepositList/LoadingList'
+import { CardTitle } from '@components/Text/CardTitle'
+import { TransactionReceiptButton } from '@components/TransactionReceiptButton'
+import { useAllChainsFilteredPromotions } from '@hooks/v4/TwabRewards/useAllChainsFilteredPromotions'
+import { useUsersRewardsHistory } from '@hooks/v4/TwabRewards/useUsersRewardsHistory'
+import { useUsersPromotionRewardsAmount } from '@hooks/v4/TwabRewards/useUsersPromotionRewardsAmount'
+import { useUsersPromotionAmountClaimable } from '@hooks/v4/TwabRewards/useUsersPromotionAmountClaimable'
+import { useUsersPromotionAmountEstimate } from '@hooks/v4/TwabRewards/useUsersPromotionAmountEstimate'
+import { capitalizeFirstLetter, transformHexColor } from '@utils/TwabRewards/misc'
+import {
+  useNextRewardIn,
+  useEstimateRows,
+  usePromotionDaysRemaining
+} from '@hooks/v4/TwabRewards/promotionHooks'
+import { getTwabRewardsContract } from '@utils/TwabRewards/getTwabRewardsContract'
+import { loopXTimes } from '@utils/loopXTimes'
+import { getAmountFromBigNumber } from '@utils/getAmountFromBigNumber'
+
+enum ClaimModalState {
+  'FORM',
+  'RECEIPT',
+  'COMPLETED'
+}
+
+export const RewardsCard = () => {
+  const { t } = useTranslation()
+
+  const promotionsQueryResults = useAllChainsFilteredPromotions()
+
+  const isFetched = promotionsQueryResults.every((queryResult) => queryResult.isFetched)
+  const isError =
+    promotionsQueryResults.map((queryResult) => queryResult.isError).filter(Boolean)?.length > 0
+  const isAny =
+    promotionsQueryResults
+      .map((queryResult) => queryResult.data?.promotions)
+      .filter((promotions) => promotions?.length > 0).length > 0
+
+  if (!isAny) {
+    return null
+  }
+
+  return (
+    <div className='flex flex-col space-y-2'>
+      <CardTitle title={t('rewards')} loading={!isFetched} />
+
+      {!isFetched && (
+        <LoadingList
+          listItems={1}
+          bgClassName='bg-pt-purple-lightest dark:bg-opacity-40 dark:bg-pt-purple'
+        />
+      )}
+
+      {isError && (
+        <div>
+          Unable to fetch rewards data due to subgraph issue, come back to check your rewards later!
+        </div>
+      )}
+
+      {promotionsQueryResults.map((queryResult) => {
+        const { data } = queryResult || {}
+        const { chainId } = data || {}
+        if (!data?.promotions || data.promotions.length === 0) {
+          return null
+        }
+        return <ChainPromotions key={`chain-promotions-${chainId}`} queryResult={queryResult} />
+      })}
+    </div>
+  )
+}
+
+const ChainPromotions = (props) => {
+  const { queryResult } = props
+
+  const { t } = useTranslation()
+
+  const { data } = queryResult
+  const { chainId, promotions } = data || {}
+
+  const backgroundColor = useNetworkHexColor(chainId)
+  const networkName = capitalizeFirstLetter(getNetworkNameAliasByChainId(chainId))
```

Comment:
`getNetworkNiceNameByChainId` returns proper names for the UI. You could also just use tailwind to capitalize the string <https://v2.tailwindcss.com/docs/text-transform#capitalize>